test/integration/model.test.js

```diff
@@ -2109,298 +2112,264 @@ describe(Support.getTestDialectTeaser('Model'), () => {
+    describe('schematic support', () => {
+      beforeEach(async function () {
+        this.UserPublic = this.sequelize.define('UserPublic', {
+          age: DataTypes.INTEGER,
+        });
+
+        this.UserSpecial = this.sequelize.define('UserSpecial', {
+          age: DataTypes.INTEGER,
+        });
+
+        await Support.dropTestSchemas(this.sequelize);
+        await this.sequelize.createSchema('schema_test');
+        await this.sequelize.createSchema('special');
+        this.UserSpecialSync = await this.UserSpecial.schema('special').sync({ force: true });
       });
 
-      this.UserSpecial = this.sequelize.define('UserSpecial', {
-        age: DataTypes.INTEGER,
+      afterEach(async function () {
+        try {
+          await this.sequelize.dropSchema('schema_test');
+        } finally {
+          await this.sequelize.dropSchema('special');
+          await this.sequelize.dropSchema('prefix');
+        }
       });
 
-      await Support.dropTestSchemas(this.sequelize);
-      await this.sequelize.createSchema('schema_test');
-      await this.sequelize.createSchema('special');
-      this.UserSpecialSync = await this.UserSpecial.schema('special').sync({ force: true });
-    });
+      it('should be able to drop with schemas', async function () {
+        await this.UserSpecial.drop();
+      });
 
-    afterEach(async function () {
-      try {
-        await this.sequelize.dropSchema('schema_test');
-      } finally {
-        await this.sequelize.dropSchema('special');
-        await this.sequelize.dropSchema('prefix');
-      }
-    });
+      it('should be able to list schemas', async function () {
+        const schemas = await this.sequelize.showAllSchemas();
 
-    it('should be able to drop with schemas', async function () {
-      await this.UserSpecial.drop();
-    });
+        const expectedSchemas = {
+          // "sequelize_test" is the default schema, which some dialects will not delete
+          mysql: ['sequelize_test', 'schema_test', 'special'],
+          mariadb: ['sequelize_test', 'schema_test', 'special'],
+          ibmi: ['sequelize_test', 'schema_test', 'special'],
+          mssql: ['schema_test', 'special'],
+          postgres: ['schema_test', 'special'],
+          db2: ['schema_test', 'special '],
+        };
 
-    it('should be able to list schemas', async function () {
-      const schemas = await this.sequelize.showAllSchemas();
-      expect(schemas).to.be.instanceof(Array);
-      const expectedLengths = {
-        mssql: 2,
-        postgres: 2,
-        db2: 2,
-        mariadb: 3,
-        mysql: 1,
-        sqlite: 1,
-        ibmi: 3,
-      };
-      expect(schemas).to.have.length(expectedLengths[dialect]);
-    });
+        expect(schemas.sort()).to.deep.equal(expectedSchemas[dialectName].sort());
+      });
 
-    if (['mysql', 'sqlite'].includes(dialect)) {
-      it('should take schemaDelimiter into account if applicable', async function () {
```

Comment:
I haven't looked through all the test files yet when making this comment, but am I correct that this test has been removed?

Reply:
Yes but it shouldn't. sqlite still uses fake schemas. I still need at least a few tests that ensure "schema" prefixes the table name (but it won't run other schema tests because they're not schemas)

Reply:
Re-added